test/test_jaches_products.jl

```diff
@@ -65,61 +65,90 @@ cache4 = ForwardDiff.Dual{typeof(ForwardDiff.Tag(Nothing, eltype(x))), eltype(x)
 @test auto_hesvecgrad!(dy, h, x, v, cache1, cache2)≈ForwardDiff.hessian(g, x) * v rtol=1e-2
 @test auto_hesvecgrad(h, x, v)≈ForwardDiff.hessian(g, x) * v rtol=1e-2
 
+# JacVec
+
 L = JacVec(f, x)
 @test L * x ≈ auto_jacvec(f, x, x)
 @test L * v ≈ auto_jacvec(f, x, v)
 @test mul!(dy, L, v) ≈ auto_jacvec(f, x, v)
-L.x .= v
+dy=rand(N);_dy=copy(dy);@test mul!(dy,L,v,a,b) ≈ a*auto_jacvec(f,x,v) + b*_dy
+update_coefficients!(L, v, nothing, 0.0)
 @test mul!(dy, L, v) ≈ auto_jacvec(f, v, v)
+dy=rand(N);_dy=copy(dy);@test mul!(dy,L,v,a,b) ≈ a*auto_jacvec(f,x,v) + b*_dy
 
 L = JacVec(f, x, autodiff = false)
 @test L * x ≈ num_jacvec(f, x, x)
 @test L * v ≈ num_jacvec(f, x, v)
-L.x == x
 @test mul!(dy, L, v)≈num_jacvec(f, x, v) rtol=1e-6
-L.x .= v
+dy=rand(N);_dy=copy(dy);@test mul!(dy,L,v,a,b) ≈ a*num_jacvec(f,x,v) + b*_dy rtol=1e-6
+update_coefficients!(L, v, nothing, 0.0)
 @test mul!(dy, L, v)≈num_jacvec(f, v, v) rtol=1e-6
+dy=rand(N);_dy=copy(dy);@test mul!(dy,L,v,a,b) ≈ a*num_jacvec(f,x,v) + b*_dy rtol=1e-6
 
-### Integration test with IterativeSolvers
 out = similar(v)
 gmres!(out, L, v)
 
-x = rand(300)
-v = rand(300)
+#=
+ff1 = ODEFunction(lorenz, jac_prototype = JacVec{Float64}(lorenz, u0))
+ff2 = ODEFunction(lorenz, jac_prototype = JacVec{Float64}(lorenz, u0, autodiff=false))
+
+for ff in [ff1, ff2]
+    prob = ODEProblem(ff, u0, tspan)
+    @test solve(prob, TRBDF2()).retcode == :Success
+    @test solve(prob, TRBDF2(linsolve = KrylovJL_GMRES())).retcode == :Success
+    @test solve(prob, Exprb32()).retcode == :Success
+    @test solve(prob, Rosenbrock23()).retcode == :Success
+    @test solve(prob, Rosenbrock23(linsolve = KrylovJL_GMRES())).retcode == :Success
+end
+=#
+
+# HesVec
```

Comment:
?

Reply:
OrdinaryDiffEq fails to precompile with this version of SparseDiffTools because of the changes we have discussed above. So I'm going to remove these tests and put them back in once we make the requisite changes in OrdianryDiffEq (in `OrdinaryDiffEq.jl/src/derivative_utils.jl`). Making a note of that here: https://github.com/SciML/SciMLOperators.jl/issues/142